public/app/features/alerting/unified/components/rule-viewer/RuleViewerVisualization.tsx

```diff
@@ -141,12 +141,11 @@ export function RuleViewerVisualization({
 function createExploreLink(settings: DataSourceInstanceSettings, model: AlertDataQuery): string {
   const { name } = settings;
   const { refId, ...rest } = model;
-  const queryParams = { ...rest, datasource: name };
 
   return urlUtil.renderUrl(`${config.appSubUrl}/explore`, {
     left: JSON.stringify({
       datasource: name,
-      queries: [{ refId: 'A', ...queryParams }],
+      queries: [{ refId: 'A', ...rest }],
```

Comment:
Is it possible that there is no `datasource` property in the `...rest`? Should we fallback in that case to `{ datasource: name }`?

Reply:
We should always have the datasource in queries. Having a datasource outside of it is legacy behavior that is currently supported but we will be moving toward removing it.

Reply:
@konrad147 yea, in my testing I've found some alerts that don't have a data source embedded inside the `model` so this could come back to us in the future when the explorer decides to make those changes.

One thing is unclear to me; why do some alert definitions not have a data source embedded in the model? Perhaps this occurs for older alerts?

I'm not sure if falling back to `{ datasource: "name" }` would help because I think we want to get rid of DS names and use UIDs everywhere.

To realy solve this without it coming back to us we should refactor and use UIDs everywhere – I don't know why Alerting is using DS names everywhere but the scope for this is quite large I imagine and might need some migrations.

Reply:
Could you add the explanation above to the PR description? Just in case someone searches for a deeper problem description in the future

Reply:
Yep, will add this before we merge